rbx_xml/src/types/numbers.rs

```diff
@@ -8,7 +8,56 @@ use crate::{
     serializer::{EncodeError, XmlWriteEvent, XmlEventWriter},
 };
 
-macro_rules! number_type {
+macro_rules! float_type {
+    ($rbx_type: ident, $type_struct: ident, $rust_type: ty, $xml_name: expr) => {
+        pub struct $type_struct;
+
+        impl XmlType<$rust_type> for $type_struct {
+            const XML_TAG_NAME: &'static str = $xml_name;
+
+            fn write_xml<W: Write>(
+                writer: &mut XmlEventWriter<W>,
+                name: &str,
+                value: &$rust_type,
+            ) -> Result<(), EncodeError> {
+                writer.write(XmlWriteEvent::start_element(Self::XML_TAG_NAME).attr("name", name))?;
+                let value = *value;
+                if value == 1.0 / 0.0 {
+                    writer.write_characters("INF")?;
+                } else if value == -1.0 / 0.0 {
+                    writer.write_characters("-INF")?;
+                } else if value.is_nan() {
+                    writer.write_characters("NAN")?;
+                } else {
+                    writer.write_characters(value)?;
+                }
+                writer.write(XmlWriteEvent::end_element())?;
+
+                Ok(())
+            }
+
+            #[allow(clippy::zero_divided_by_zero)]
```

Comment:
Neither std::$rust_type::NAN nor std::<$rust_type>::NAN compile, so 0.0 / 0.0 is needed.

Reply:
I'd expect `std::<$rust_type>::NAN` to compile, interesting.

Reply:
at least there's a bug for it: https://github.com/rust-lang/rust/issues/48067

Reply:
I think this isn't bad enough for readability to matter?

Reply:
Yeah, it's proabbly fine.